test/ha/activator_test.go

```diff
@@ -0,0 +1,146 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ha
+
+import (
+	"fmt"
+	"log"
+	"sort"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	revisionresourcenames "knative.dev/serving/pkg/reconciler/revision/resources/names"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+)
+
+const (
+	activatorDeploymentName = "activator"
+	activatorLabel          = "app=activator"
+)
+
+// The Activator is managed by the activator HPA and does not have leader election enabled.
+// The test ensures that stopping one of the activator pods doesn't affect user applications.
+// One service is probed during activator restarts and another service is used for testing
+// that we can scale from zero after activator restart.
+func TestActivatorHA(t *testing.T) {
+	clients := e2e.Setup(t)
+
+	pods, err := clients.KubeClient.Kube.CoreV1().Pods(servingNamespace).List(metav1.ListOptions{
+		LabelSelector: activatorLabel,
+	})
+	if err != nil {
+		t.Fatalf("Failed to get activator pods: %v", err)
+	}
+	// make sure we have the right number of activator replicas before continuing
+	if len(pods.Items) != haReplicas {
+		t.Skip(fmt.Sprintf("The test requires %d activator pods, got: %d", haReplicas, len(pods.Items)))
```

Comment:
```suggestion
		t.Skipf("The test requires %d activator pods, got: %d", haReplicas, len(pods.Items))
```

Reply:
Also don't we want to poll here until we get the right number?

Reply:
This is ensured by the HPA for activator which is enabled in the Bash scripts before any tests run. So, if we keep the HPA we don't need to wait. We should have the right number at this moment.

Reply:
Added the check anyway